.github/workflows/pr-scan.yml

```diff
@@ -1,8 +1,8 @@
 name: PR Security Scan
 on:
-  pull_request:
-    types: [opened, synchronize, reopened]
-  workflow_dispatch:
+  pull_request_target:
+    branches: ['**']
```

Comment:
this allows to do lots of permissions that might not be there 🤔 

I have not found a good way on how to handle these scenarios :-/

Reply:
jobs triggered from pull_request_target are executed in the target branch. Since will be using the workflow file in the target branch, it won't be possible for contributors to modify the workflow to do something malicious.

Source: https://stackoverflow.com/questions/74957218/what-is-the-difference-between-pull-request-and-pull-request-target-event-in-git

If anything, there may be concern with secrets being in repo being dumped, but we will need to mitigate that by reading the code and making sure nothing nefarious is being done.

Besides that, we can limit the permissions on the `GITHUB_TOKEN` and move secrets to a separate environment to be used only when we deploy.